### DISCUSSION TITLE
units system

### DISCUSSION THREAD

**Discussion by rh201** [idaholab/moose#19256] [Q&A General] [answered]
Hello, 

If I generate mesh in the unit of mm rather than m, then how should the units system in the input file be changed?

For example, when I use m in the mesh and input density = 1kg/m^3, thermal conductivity = 1W/mK and specific heat = 1 J/kgK , then what's the unit of them in the input file when the mesh with unit of mm ?

I checked the page https://mooseframework.inl.gov/source/utils/Units.html  but find no solutions. Thank you:)

**Comment by GiudGiud** [comment, marked as answer]
Hello

If the mesh is in mm then all units need to be adapted to mm. I recommend you document them throughout your input file. If you use any pre-made model (like the fluid properties from the fluid_properties module), make sure to check those for units too. 

So in the input file:
1kg / m3 -> 1e-9 kg/mm3 now
1W/mk -> 1e-3 W/mmK
etc

That page is one of the best resource wrt unit conversions. The other discussion about how to do units in input files is here
https://mooseframework.inl.gov/application_usage/input_syntax.html

Guillaume

**Reply by rh201** [reply]
@GiudGiud Thank you:)
